pages/index.tsx

```diff
@@ -0,0 +1,38 @@
+import React, { useState } from "react";
+import api from "../services/axios";
+
+export default function Produtos() {
+  const [productPrice, setProductPrice] = useState(0);
+  const [phones, setPhones] = useState<Phone[]>([]);
+
+  const handleChangePrice = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const price = parseInt(e.target.value) || 0;
+    setProductPrice(price);
+  };
+
+  const handleSearch = async () => {
+    try {
+      const results = await api.get<Phone[]>("phones", {
+        params: { price: productPrice },
+      });
+      results.data && setPhones(results.data);
+    } catch (err) {
+      console.log("deu ruim amigao", err);
```

Comment:
Create a better error message. You can create an alert or something like this when you catch errors connecting with the API.